tests/Fortran-tests/io_matrix_typed.F90

```diff
@@ -29,9 +33,9 @@ function test_io_matrix_typed(matrix_type, element_kind, element_precision, n, m
     DUMMY_KIND(DUMMY_PREC), allocatable :: a_dense(:, :)
     DUMMY_KIND(DUMMY_PREC), allocatable :: b_dense(:, :)
 
-    pid = getpid()
-    write(pid_char,*)pid
-    write(fname,*)"ftest_matrix",trim(adjustl(pid_char)),".mtx"
+    call date_and_time(date,time,zone,values)
```

Comment:
I am not very happy with this change. What about syncing it with the C version and using `mkstemp()`?

Reply:
Are you suggesting to create a Fortran interface for mkstemp() and change the call from mktemp() to mkstemp() in the C code? Or is mkstemp() already available in fortran.

Reply:
Hmm, good point. I don't think this function is readily available from Fortran. Also, it might be tricky to make this available in Fortran because `mkstemp()` returns an open file descriptor which I am not sure how to use in Fortran. Ok, let's forget about this then for now and leave it as you proposed. Sorry for the confusion.

Reply:
Do you have any objection to restoring the getpid() and using, e.g.

#ifdef INTEL_SDK

to control whether to USE IFPORT?

We could then define INTEL_SDK or equivalent macro in the Intel compiler flags in CMakeLists.txt.

On the other hand, I don't see any other such macros currently in the fortran code, so maybe avoiding them is important for other reasons.

Reply:
No, that would work as well. It would only affect one (test) source which is completely fine in my opinion.

Reply:
Done. Please check.